repl/src/main/scala/ammonite/repl/interp/Interpreter.scala

```diff
@@ -45,7 +51,34 @@ class Interpreter(handleResult: => Res[Evaluated] => Unit,
         printer
       )
     } finally Thread.currentThread().setContextClassLoader(oldClassloader)
-  } yield out
+  }
+
+  def processScript(code: String) = {
+    val preprocessed = preprocess.script(Parsers.splitScript(code),eval.getCurrentLine)
+    preprocessed match {
+      case Res.Failure(msg) =>
+        stdout(Console.RED + msg + Console.RESET + "\n")
+      case Res.Success(blocks) =>
+        loop(blocks)
+    }
+
+    @tailrec def loop(blocks: Seq[Preprocessor.Output]){
+      if(!blocks.isEmpty){
+        val Preprocessor.Output(code, printSnippet) = blocks.head
+        val ev = evaluateLine(code, printSnippet, _ => ())
+        ev match {
+          case Res.Success(ev) =>
+            eval.update(ev.imports)
+          case _ => Unit
```

Comment:
Wait what's this? Did you want `()`? Regardless, you should be able to just leave it blank